### ISSUE TITLE
RFC: remove protobuf support from write_prometheus?

### ISSUE THREAD

**Issue by rubenk** [collectd/collectd#3106]
It hasn't been supported since Prometheus 2.0, released in November 2017 (https://github.com/prometheus/docs/issues/927

**Comment by trustchk** [comment]
I'm very much in favor of this, as I'm having a hard time compiling `protobuf` and `protobuf-c` on AIX.
Using `write_prometheus` would allow us to use a standard setup where prometheus scrapes the output, instead of having to rely on `collectd_exporter` on linux nodes which would make matters complicated for us.

**Comment by eero-t** [comment]
I would also be in favor of simplifying the plugin.  Something that is nowadays used just to generate (Prometheus) ASCII output, shouldn't look that complicated.

**Comment by eero-t** [comment]
Above "write_prometheus: migration to v6.0" PR which removes protobuf support has been merged, so this can be closed.

**Comment by elfiesmelfie** [comment]
@eero-t: is that support also removed from collectd 5.x?
IMHO, it should be backported to main, since collectd-6.0 isn't released yet (while collectd 5.13 probably isn't going to be released any time soon, at least having it on main will lesson the effort should people wish to backport it in various distros).

**Comment by eero-t** [comment]
> is that support also removed from collectd 5.x?

@elfiesmelfie No, nobody has proposed similar PR for "main" branch.  I've already switched to v6 branch and would suggest that to anybody else who's building collectd from source, hacking it, or using write_prometheus plugin, to do the same.

> IMHO, it should be backported to main, since collectd-6.0 isn't released yet.

According to previous schedule, v6 should have already been released, and v5 in maintenance mode.  Unfortunately due to the earlier project stalls, most of the plugins haven't been ported yet, although quite a few of them have open migration PRs.  

I would actually propose adopting a policy that all PRs should go v6 first, and only once they are there, backport PRs to v5 would be accepted.  Bug fixes could be accepted to v5 first, but v6 PR should be present before that, or there should be argument about why given fix is not needed for v6.  Otherwise v6 will be in limbo forever because:
* nobody pushes things forward in v6 branch because all development happens in v5
* plugins have moved too much forward in v5 branch, and switching would regress the plugins

IMHO that policy should have been already there for all the plugins that have already been migrated to v6, or have a PR for that...

> (while collectd 5.13 probably isn't going to be released any time soon, at least having it on main will lesson the effort should people wish to backport it in various distros).

Many things require / depend on protobuf, so having that dependency also in collectd should not be a problem for distros, only for some marginal OSes, which are lacking also a lot of other SW.  If somebody needs that support to be dropped from v5, they should propose PR for it.  That should be easy to do based on the v6 one.

(I'm not going to do that as I'm already on v6.)
